examples/developer-tools/src/blocks/input.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly/core';
+
+/**
+ * Check to see if more than one input has this name.
+ * If so, applies a warning to the block.
+ * 
+ * Highly inefficient (On^2), but n is small.
+ * @param referenceBlock Block to check.
+ */
+const inputNameCheck = function (referenceBlock: Blockly.Block) {
+  if (referenceBlock.isDeadOrDying()) {
+    // Block has been deleted.
+    return;
+  }
+  const name = referenceBlock.getFieldValue('INPUTNAME').toLowerCase();
+  let count = 0;
+  const blocks = referenceBlock.workspace.getAllBlocks(false);
+  for (const block of blocks) {
+    const otherName = block.getFieldValue('INPUTNAME');
+    if (
+      block.isEnabled() &&
+      !block.getInheritedDisabled() &&
+      otherName &&
+      otherName.toLowerCase() === name
+    ) {
+      count++;
+    }
+  }
+  const msg =
+    count > 1 ? 'There are ' + count + ' input blocks\n with this name.' : null;
+  referenceBlock.setWarningText(msg);
+};
+
+// Inputs that should take a "type" input for connection checks
+const inputsWithTypeInputs = new Set(['input_value', 'input_statement']);
+
+const inputTypeValidator = function (
+  this: Blockly.Block,
+  value: string,
+): string {
+  if (inputsWithTypeInputs.has(value)) {
+    if (!this.getInput('TYPE')) {
+      this.appendValueInput('TYPE')
+        .setCheck(null)
+        .setAlign(Blockly.inputs.Align.RIGHT)
+        .appendField('type');
+    }
+  } else {
+    this.removeInput('TYPE', true);
+  }
+
+  return value;
+};
+
+const tooltip: Record<string, string> = {
+  input_value: 'A value socket for horizontal connections.',
+  input_statement: 'A statement socket for enclosed vertical stacks.',
+  input_dummy:
+    'For adding fields without any block connections. Alignment options (left, right, centre) only affect multi-row blocks.',
+  input_end_row:
+    'For adding fields without any block connections that will be rendered on a separate row from any following inputs. Alignment options (left, right, centre) only affect multi-row blocks.',
```

Comment:
Prettier doesn't change the line length of long strings, and again, if prettier doesn't format something I don't think we should do it manually.

Reply:
I agree that we shouldn't do things that prettier doesn't do. But prettier is opposed to wrapping strings (see https://github.com/prettier/prettier/issues/4123). So we're making a decision that long strings are ok. I don't super love that because I think it'll make unit tests hard to review. But not the biggest deal to me either.